mlir/lib/Dialect/Rock/IR/AmdArchDb.cpp

```diff
@@ -77,32 +90,252 @@ static constexpr AmdArchInfo
                   GemmFeatures::atomic_fmax_f32 | GemmFeatures::wmma,
-              /*waveSize=*/32, /*maxWavesPerEU*/ 20, /*totalSGPRPerEU*/ 512,
+              /*waveSize=*/32, /*maxWavesPerEU*/ 16, /*totalSGPRPerEU*/ 800,
+              /*totalVGPRPerEU*/ 1536, /*totalSharedMemPerCU*/ 131072,
+              /*maxSharedMemPerWG*/ 65536, /*numEUPerCU=*/4, /*minNumCU=*/12,
+              /*hasFp8ConversionInstrs=*/false,
+              /*hasOcpFp8ConversionInstrs=*/false, /*maxNumXCC=*/1),
+    rdna4Info(GemmFeatures::dot | GemmFeatures::atomic_add |
+                  GemmFeatures::atomic_fmax_f32 | GemmFeatures::wmma |
+                  GemmFeatures::atomic_add_f16 | GemmFeatures::atomic_add_bf16,
+              /*waveSize=*/32, /*maxWavesPerEU*/ 16, /*totalSGPRPerEU*/ 800,
               /*totalVGPRPerEU*/ 1536, /*totalSharedMemPerCU*/ 131072,
               /*maxSharedMemPerWG*/ 65536, /*numEUPerCU=*/4, /*minNumCU=*/12,
               /*hasFp8ConversionInstrs=*/false,
-              /*hasOcpFp8ConversionInstrs=*/false, /*maxNumXCC=*/1);
+              /*hasOcpFp8ConversionInstrs=*/true, /*maxNumXCC=*/1);
+
+namespace {
+
+template <typename LHS, typename RHS>
+std::enable_if_t<std::is_assignable_v<LHS &, RHS &&>, void>
+checkAndSetInfo(StringRef name, LHS &lhs, RHS &&rhs) {
+  if (lhs != static_cast<LHS>(rhs)) {
+    LLVM_DEBUG(llvm::dbgs() << "NOTE: Value discrepancy for " << name << ": "
+                            << lhs << " (old) != " << rhs
+                            << " (new). Proceeding with " << rhs << ".\n");
+    lhs = std::forward<RHS>(rhs);
+  }
+}
+
+std::tuple<StringRef, unsigned> parseArchString(StringRef arch) {
+  std::tuple<StringRef, unsigned> ret("", 0);
 
-AmdArchInfo mlir::rock::lookupArchInfo(StringRef arch) {
-  // Keep this implementation in sync with
-  // mlir/test/lit.site.cfg.py.in:set_arch_features()
   StringRef firstPart, remainingParts;
   std::tie(firstPart, remainingParts) = arch.split(':');
-  if (firstPart.contains('-')) { // target triple
-    std::tie(firstPart, remainingParts) = remainingParts.split(':');
+  if (firstPart == "native") {
+    std::get<0>(ret) = firstPart;
+    if (unsigned long long deviceId;
+        !llvm::getAsUnsignedInteger(remainingParts, 0, deviceId)) {
+      std::get<1>(ret) = deviceId;
+    }
+  } else {
+    auto chipPos = firstPart.find("gfx");
+    if (chipPos != StringRef::npos) {
+      firstPart = firstPart.substr(chipPos);
+    } else {
+      std::tie(firstPart, remainingParts) = remainingParts.split(':');
+    }
+    std::get<0>(ret) = firstPart;
+  }
+
+  return ret;
+}
+
+struct AgentInfo {
+  // Input fields:
+  //   The ID of the GPU device that we are looking for.
+  unsigned deviceId;
+  //   Used in acquireAgentInfo, to compute GPU internal IDs.
+  int numCpus;
+  // Output fields:
+  uint32_t simdsPerCU;
+  uint32_t maxWavesPerCU;
+};
+
+AmdArchInfo fetchNativeArchInfo(const hipDeviceProp_t &prop,
```

Comment:
Can we move that to the migraphx_gpu target instead? To read the required properties there and pass them to the rocMLIR instead of makeing HIP a dependency of rocMLIR?

Currently the HIP dependency compilcates a lot on Windows for rocMLIR.